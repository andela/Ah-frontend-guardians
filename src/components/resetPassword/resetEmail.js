import React, { Component } from 'react';
import { NavBar } from '../navBar/NavBar';
import '../../css/resetPassword.css';
import axios from 'axios';
import fetchResetEmail from '../../actions/resetPassword/resetPasswordAction';
import { connect } from 'react-redux';
import toast from 'react-toastify';


export class ResetEmail extends Component {
    constructor(props){
        super(props)
        this.state = {}
    }
    onSubmit = (e) => {
        e.preventDefault()
        
        this.props.fetchResetEmail(this.state.reset_email)
    }

    onChange = event => {
        this.setState({ [event.target.name]: event.target.value });
        
    };

    render() {
        return (
        <div className="reset-email reset">
            <a href="/"><h1>Author's Haven</h1></a>
            <p>Enter your email address here and an email will be sent to you for you to complete this request.</p>
            <div className="reset-email-div">
                <form className="email-form" onSubmit={this.onSubmit}>
                    <input type="email" onChange={this.onChange} placeholder="Enter email . . ." name="reset_email" id="reset_email" required/><br /><br />
                    <input type="submit" value="Reset" id="submit-btn"/><br /><br />
                    <a href="/login">Cancel</a>
                </form>
            </div>
        </div>
        )
    }
}

export const mapStateToProps = (state) => {
    return state.resetEmailReducer
}

export default connect(mapStateToProps, { fetchResetEmail })(ResetEmail);
