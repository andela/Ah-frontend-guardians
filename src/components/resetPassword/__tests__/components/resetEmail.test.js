import React from 'react';
import { shallow, mount } from 'enzyme';
import { ResetEmail, mapStateToProps } from '../../resetEmail';

describe('Reset Email Test', () => {
    const wrapper = shallow(<ResetEmail/>)

    const mockfetchResetPasswordfn = jest.fn()

    const props ={
        fetchResetEmail: mockfetchResetPasswordfn,
        message: 'Password Reset'
    }

    test('renders the component without crashing', () => {

        expect(wrapper).toMatchSnapshot()
    });

    test("should map states to props", () => {
        const mockedState = {
            resetEmailReducer: {sent: false}
        }
        const state = mapStateToProps(mockedState);
        expect(state).toEqual({sent: false});
    });

    test('should render without crashing', () => {
        const component = mount(<ResetEmail/>)
        expect(component.find('.reset-email').length).toEqual(1);
    });

    test('It should sunmit', () =>{
        const component = shallow(<ResetEmail {...props} />)
        component.setState({
            sent: false
        })
        component.instance().onSubmit({preventDefault() {} })
        expect(component.state()).toEqual({sent: false});
    });

    test('should call onChange', () => {
        const onChangeMock = jest.fn();
        const event = {
          target: { name: 'reset_email', value: 'barnabastb2@gmail.com' }
        };
        const component = shallow(<ResetEmail onChange={onChangeMock} />);
        component.instance().onChange(event)
        expect(component.state('reset_email')).toEqual('barnabastb2@gmail.com');
    });
});
