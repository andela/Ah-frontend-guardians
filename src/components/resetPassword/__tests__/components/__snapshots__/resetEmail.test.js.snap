// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Email Test renders the component without crashing 1`] = `ShallowWrapper {}`;
